atc/engine/engine.go

```diff
@@ -22,7 +22,7 @@ import (
 
 type Engine interface {
 	NewBuild(db.Build) Runnable
-	NewCheck(db.Check) Runnable
+	NewCheck(context.Context, db.Check) Runnable
```

Comment:
I recognize this would result in a slightly bigger refactor, but is there a reason we pass the `Context` in here as opposed to modifying `Runnable` interface to take in the `Context`? To quote the [godocs](https://godoc.org/context) for the `context` package:

> Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it.

Correct me if I'm wrong, but we always seem to call `NewBuild` and `NewCheck` immediately adjacent to `Run`, so it shouldn't matter which function we pass it into (in terms of what context we have available)